src/variational/VariationalInference.jl

```diff
@@ -95,15 +92,17 @@ end
 function grad!(
     vo,
     alg::VariationalInference{<:ForwardDiffAD},
-    q::VariationalPosterior,
-    model::Model,
+    q,
+    model,
     θ::AbstractVector{<:Real},
     out::DiffResults.MutableDiffResult,
     args...
 )
-    # TODO: this probably slows down executation quite a bit; exists a better way
-    # of doing this?
-    f(θ_) = - vo(alg, q, model, θ_, args...)
+    f(θ_) = if (q isa VariationalPosterior)
```

Comment:
Can we implement a function for this and achive this `if` statement via multiple dispatch?

Reply:
TL;DR: Tried it and it was super-slow for some reason. Current approach is alright IMO, and think we can explore making it nicer in AdvancedVI.jl instead:)

Last week or so I implemented this by adding a "wrap" function, e.g. something along the lines of
```julia
function wrapped_objective(...)
    function f(...)
        # define objective
    end

    return f
end
```
But I ended up with a 10X slow-down because of this, and so I figured "ehh, I'll leave this for now and we can explore this further in AdvancedVI.jl". Also because I wasn't quite happy with the outcome; it's difficult to make this "wrap" approach work in general, i.e. if someone implements a custom `grad!` it's very likely they'll also have to implement their custom `wrap_objective` due to arguments and whatnot.

Reply:
Yes let's defer it to AVI

Reply:
Great! Then I'll address the other comments and it should be ready to merge

Reply:
Yes sure. Please ping me when it's ready to merge.